src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -420,6 +440,26 @@ static int _PaPulseAudio_ProcessAudio(PaPulseAudio_Stream *stream,
             PaUtil_SetOutputFrameCount( &stream->bufferProcessor,
                                         hostFramesPerBuffer );
 
+        }
+
+        hostFrameCount =
+            PaUtil_EndBufferProcessing( &stream->bufferProcessor,
+                                        &ret );
+
+        PaUtil_EndCpuLoadMeasurement( &stream->cpuLoadMeasurer,
+                                      hostFrameCount );
+
+        /*
+         * Idea behind this is that is there is output we have
```

Comment:
Not sure whether this is what you meant:

```
         * The idea behind this is that if we have allocated
         * an output buffer with pa_stream_begin_write
         * and then if PaUtil_EndBufferProcessing returns
         * paContinue we write data to the buffer, but if
         * there is no further output (ret is something
         * other than paContinue) then we have to cancel
         * the write with pa_stream_cancel_write
```

But I don't think this behavior is correct. Please check other host APIs. I think you should be calling pa_stream_write whenever hostFrameCount > 0.

Reply:
I checked ALSA and it's done the same. It does not even check `hostFrameCount` only if return in something else than `paContinue`